llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -2079,12 +2082,27 @@ MemoryDepChecker::getDependenceDistanceStrideAndSize(
   if (StrideAScaled == StrideBScaled)
     CommonStride = StrideAScaled;
 
+  // If the distance is a SCEVCouldNotCompute, fail immediately.
+  if (isa<SCEVCouldNotCompute>(Dist)) {
+    LLVM_DEBUG(dbgs() << "LAA: Dependence because of uncomputable distance.\n");
+    return Dependence::Unknown;
+  }
```

Comment:
Is this directly related to support different sizes?

Reply:
Thanks, split out https://github.com/llvm/llvm-project/pull/148841.